lib/libnvpair/libnvpair_json.c

```diff
@@ -84,7 +84,7 @@ nvlist_print_json_string(FILE *fp, const char *input)
 			FPRINTF(fp, "\\b");
 			break;
 		default:
-			if ((c >= 0x00 && c <= 0x1f) ||
+			if (((int)c >= 0x00 && c <= 0x1f) ||
```

Comment:
Since `c` is `wchar_t`, you should cast to `wint_t`, right?

Reply:
yes, that's better; I'm not big wchar.h enjoyer :0

Reply:
I'm accepting patches for system_data_types.7 :)

Reply:
heh:
```
libnvpair_json.c: In function 'nvlist_print_json_string':
libnvpair_json.c:87:19: error: comparison of unsigned expression >= 0 is always true [-Werror=type-limits]
   87 |    if (((wint_t)c >= 0x00 && c <= 0x1f) ||
      |                   ^~
```

I'm just gonna upcast to `long long` instead

Reply:
lol.  I guess it's ARM that defines it as unsigned.  In the end it's better to silence the warning directly.  I prefer that over casting.